pvlib/iotools/midc.py

```diff
@@ -0,0 +1,105 @@
+"""Functions to read NREL MIDC data.
+"""
+from functools import partial
+import pandas as pd
+
+# VARIABLE_MAP is a dictionary mapping partial MIDC field names to their
+# pvlib names. See docstring of read_midc for description.
+
+VARIABLE_MAP = {
+    'Direct': 'dni',
+    'Global': 'ghi',
+    'Diffuse': 'dhi',
+    'Airmass': 'airmass',
+    'Azimuth Angle': 'solar_azimuth',
+    'Zenith Angle': 'solar_zenith',
+    'Air Temperature': 'temp_air',
+    'Temperature': 'temp_air',
+    'Dew Point Temp': 'temp_dew',
+    'Relative Humidity': 'relative_humidity',
+}
+
+
+def map_midc_to_pvlib(variable_map, field_name):
+    """A mapper function to rename Dataframe columns to their pvlib counterparts.
+
+    Parameters
+    ----------
+    variable_map: Dictionary
+        A dictionary for mapping MIDC field nameto pvlib name. See VARIABLE_MAP
+        for default value and description of how to construct this argument.
+    field_name: string
+        The Column to map.
+
+    Returns
+    -------
+    label: string
+        The pvlib variable name associated with the MIDC field or the input if
+        a mapping does not exist.
+    """
+    new_field_name = field_name
+    for midc_name, pvlib_name in variable_map.items():
+        if field_name.startswith(midc_name):
+            # extract the instument and units field and then remove units
+            instrument_units = field_name[len(midc_name):]
+            instrument = instrument_units[:instrument_units.find('[') - 1]
```

Comment:
I think this logic assumes that units are always specified in this format and will produce the wrong result otherwise. Probably a safe assumption, but maybe add a comment.

Reply:
This is something I had considered and forgot to include in implementation. Would it be preferable to handle the case where no units are provided? e.g. instrument_units.find('[') returns -1?